### PR TITLE
[UI] As a user, I can have an Action bar to navigate to All Filters, History, and Logout screens

### LINKED ISSUE #31: [UI] As a user, I can back to Home screen from the Add Filter screen
## Why

Display Back button on the toolbar to back to the previous screen
 
## Acceptance Criteria

- Display on the Add Filter screen
- Don't need to display on Home screen
- Click on the Back button will process the same as click on the Back on system Navigation bar

### PR DESCRIPTION
**Pull request by lydiasama**
## What happened 👀

Add toolbar with side modal and back button which handle by the [NavigationUI](https://developer.android.com/reference/kotlin/androidx/navigation/ui/NavigationUI#setupactionbarwithnavcontroller_1)
 
## Insight 📝

- The title of the toolbar will get from the `android:label` in nav_graph.
- If it is the top-level destination, the left icon will be a hamburger icon. Otherwise, it will be a back one.
 
## Proof Of Work 📹

https://user-images.githubusercontent.com/12026942/127345174-df42ecd5-1a12-4700-a669-9ee0aef09738.mp4

### CODE REVIEW

**Comment by hoangnguyen92dn** [comment]
@lydiasama The rest of things is handling back action on the App bar, right? 🤔

**Comment by lydiasama** [comment]
@hoangnguyen92dn @luongvo This feature is finished. 🙌🏻

**Comment by hoangnguyen92dn** [comment]
@lydiasama When I click on the Filter the app crashed

https://user-images.githubusercontent.com/6950766/127354976-08501001-8d86-4ed3-9514-340c20591d62.mp4

And I'd suggest adding the app version at the bottom of the navigation drawer



What do you think about update the App Bar color a bit to match the status bar color or the orange as the background? 🤔

**Comment by lydiasama** [comment]
@hoangnguyen92dn I cannot reproduce the crash. It works well with my both emulator and the real device.
For the status bar color and the version name at the bottom of the navigation modal, I think we can do it later. 🙆🏻‍♀️

File: android/app/src/main/res/values/colors.xml [hunk 1]
```diff
@@ -5,9 +5,6 @@
     <color name="colorPrimaryDark">@color/purple_violent_violet</color>
     <color name="colorAccent">@color/red_alizarin</color>
 
-    <color name="colorStatusBar">@null</color>
-    <color name="colorNavigationBar">@null</color>
```
Comment (hoangnguyen92dn) [inline android/app/src/main/res/values/colors.xml]: I think we should put it back, we might need to support full screen if needed

File: android/app/src/main/res/values/styles.xml [hunk 2]
```diff
@@ -3,9 +3,8 @@
     <!-- Base application theme. -->
     <style name="AppTheme" parent="Theme.MaterialComponents.NoActionBar">
         <!-- Customize your theme here. -->
-        <item name="android:statusBarColor">@color/colorStatusBar</item>
-        <item name="android:navigationBarColor">@color/colorNavigationBar</item>
```
Comment (hoangnguyen92dn) [inline android/app/src/main/res/values/styles.xml:7]: Same here. Please put it back 😅

**Comment by lydiasama** [comment]
@hoangnguyen92dn Updated.

**Review by hoangnguyen92dn — APPROVED** [review]

**Pull request merged**